extension: cert-manager acme-resources Order / Challenge using wrong value in @Group Annotation

### Describe the bug

Following cert-manager acme classes are using the wrong k8s api group: `cert-manager.io` instead of `acme.cert-manager.io`
* [Order](https://github.com/fabric8io/kubernetes-client/blob/master/extensions/certmanager/model-v1/src/generated/java/io/fabric8/certmanager/api/model/acme/v1/Order.java#L73)
* [Challenge](https://github.com/fabric8io/kubernetes-client/blob/master/extensions/certmanager/model-v1/src/generated/java/io/fabric8/certmanager/api/model/acme/v1/Challenge.java#L73)

This leads to empty results using `Order` or `Challenge` using current cert-manager extension.

Using kubectl requesting those resources with api-version `cert-manager.io` throws errors:
* `kubectl get orders.v1.cert-manager.io`
* `kubectl get challenges.v1.cert-manager.io`

Using kubectl requesting those resources with api-version `acme.cert-manager.io` works:
* `kubectl get orders.v1.acme.cert-manager.io`
* `kubectl get challenges.v1.acme.cert-manager.io`


### Fabric8 Kubernetes Client version

6.3.1

### Steps to reproduce

Running this java code throws in both cases exceptions
```java
            OrderList orders = certManagerClient.v1().orders().list();
            if(orders.getItems().isEmpty()) {
                throw new IllegalStateException("No orders found!");
            }
```

```java

            ChallengeList challenges = certManagerClient.v1().challenges().list();
            if(challenges.getItems().isEmpty()) {
                throw new IllegalStateException("No challenges found!");
            }
```


### Expected behavior

Using the correct value in `@Group`-annotation with `acme.cert-manager.io` for `Order` and `Challenge` returns existing resources

### Runtime

Kubernetes (vanilla)

### Kubernetes API Server version

other (please specify in additional context)

### Environment

Linux

### Fabric8 Kubernetes Client Logs

_No response_

### Additional context

_No response_

@rohanKanojia running make fails 

```
make
go mod vendor
go: downloading k8s.io/apiextensions-apiserver v0.24.2
go: downloading github.com/cert-manager/cert-manager v1.9.0-beta.1.0.20220829113803-8465f1223efb
go: downloading k8s.io/apimachinery v0.24.2
go: downloading k8s.io/klog v1.0.0
go: downloading k8s.io/utils v0.0.0-20220210201930-3a6ce19ff2f9
go: downloading github.com/gogo/protobuf v1.3.2
go: downloading sigs.k8s.io/structured-merge-diff/v4 v4.2.1
go: downloading sigs.k8s.io/json v0.0.0-20211208200746-9f7c6b3444d2
go: downloading github.com/google/gofuzz v1.2.0
go: downloading k8s.io/klog/v2 v2.70.0
go: downloading gopkg.in/yaml.v2 v2.4.0
go: downloading golang.org/x/net v0.0.0-20220802222814-0bcc04d9c69b
go: downloading github.com/json-iterator/go v1.1.12
go: downloading github.com/go-logr/logr v1.2.2
go: downloading sigs.k8s.io/gateway-api v0.5.0
go: downloading k8s.io/api v0.24.2
go: downloading github.com/modern-go/reflect2 v1.0.2
go: downloading golang.org/x/text v0.3.7
CGO_ENABLED=0 GO111MODULE=on GO15VENDOREXPERIMENT=1 go run -mod=vendor -a ./cmd/generate/generate.go > ../model-v1/src/main/resources/schema/certmanager-schema.json
# sigs.k8s.io/json/internal/golang/encoding/json
vendor/sigs.k8s.io/json/internal/golang/encoding/json/encode.go:1249:12: sf.IsExported undefined (type reflect.StructField has no field or method IsExported)
vendor/sigs.k8s.io/json/internal/golang/encoding/json/encode.go:1255:18: sf.IsExported undefined (type reflect.StructField has no field or method IsExported)
make: *** [Makefile:28: gobuild] Error 2
```

so I switched to master branch run make again and even there it fails 
```
go mod vendor
CGO_ENABLED=0 GO111MODULE=on GO15VENDOREXPERIMENT=1 go run -mod=vendor -a ./cmd/generate/generate.go > ../model-v1/src/main/resources/schema/certmanager-schema.json
# sigs.k8s.io/json/internal/golang/encoding/json
vendor/sigs.k8s.io/json/internal/golang/encoding/json/encode.go:1249:12: sf.IsExported undefined (type reflect.StructField has no field or method IsExported)
vendor/sigs.k8s.io/json/internal/golang/encoding/json/encode.go:1255:18: sf.IsExported undefined (type reflect.StructField has no field or method IsExported)
make: *** [Makefile:28: gobuild] Error 2
```

running on ubuntu 22.04 and go version `go version go1.15.15 linux/amd64`

Regarding a test to verify group has been update: do you mean adding a unit-test here https://github.com/grafjo/kubernetes-client/tree/master/extensions/certmanager/tests ?

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -4,6 +4,8 @@
 
 #### Bugs
 
+* Fix #4668: use acme.cert-manager.io ApiGroup for Orders and Challenges
+
 #### Improvements
 * Fix #4637: all pod operations that require a ready / succeeded pod may use withReadyWaitTimeout, which supersedes withLogWaitTimeout.
 * Fix #4633: provided inline access to all RunConfig builder methods via run().withNewRunConfig()

--- a/extensions/certmanager/generator-v1/cmd/generate/generate.go
+++ b/extensions/certmanager/generator-v1/cmd/generate/generate.go
@@ -58,7 +58,7 @@ func main() {
 	// optional ApiGroup and ApiVersion for the go package (which is added to the generated java class)
 	packageMapping := map[string]schemagen.PackageInformation{
 		"github.com/cert-manager/cert-manager/pkg/apis/certmanager/v1": {JavaPackage: "io.fabric8.certmanager.api.model.v1", ApiGroup: "cert-manager.io", ApiVersion: "v1"},
-		"github.com/cert-manager/cert-manager/pkg/apis/acme/v1":        {JavaPackage: "io.fabric8.certmanager.api.model.acme.v1", ApiGroup: "cert-manager.io", ApiVersion: "v1"},
+		"github.com/cert-manager/cert-manager/pkg/apis/acme/v1":        {JavaPackage: "io.fabric8.certmanager.api.model.acme.v1", ApiGroup: "acme.cert-manager.io", ApiVersion: "v1"},
 		"github.com/cert-manager/cert-manager/pkg/apis/meta/v1":        {JavaPackage: "io.fabric8.certmanager.api.model.meta.v1", ApiGroup: "cert-manager.io", ApiVersion: "v1"},
 		"github.com/cert-manager/cert-manager/internal/apis/acme":      {JavaPackage: "io.fabric8.certmanager.api.model.acme.internal", ApiGroup: "cert-manager.io", ApiVersion: "v1"},
 		"github.com/cert-manager/cert-manager/internal/apis/meta":      {JavaPackage: "io.fabric8.certmanager.api.model.meta.internal", ApiGroup: "cert-manager.io", ApiVersion: "v1"},

--- a/extensions/certmanager/model-v1/src/generated/java/io/fabric8/certmanager/api/model/acme/v1/Challenge.java
+++ b/extensions/certmanager/model-v1/src/generated/java/io/fabric8/certmanager/api/model/acme/v1/Challenge.java
@@ -70,7 +70,7 @@ import lombok.experimental.Accessors;
     @TemplateTransformation(value = "/manifest.vm", outputPath = "META-INF/services/io.fabric8.kubernetes.api.model.KubernetesResource", gather = true)
 })
 @Version("v1")
-@Group("cert-manager.io")
+@Group("acme.cert-manager.io")
 public class Challenge implements HasMetadata, Namespaced
 {
 
@@ -80,7 +80,7 @@ public class Challenge implements HasMetadata, Namespaced
      * 
      */
     @JsonProperty("apiVersion")
-    private String apiVersion = "cert-manager.io/v1";
+    private String apiVersion = "acme.cert-manager.io/v1";
     /**
      * 
      * (Required)

--- a/extensions/certmanager/model-v1/src/generated/java/io/fabric8/certmanager/api/model/acme/v1/ChallengeList.java
+++ b/extensions/certmanager/model-v1/src/generated/java/io/fabric8/certmanager/api/model/acme/v1/ChallengeList.java
@@ -73,7 +73,7 @@ import lombok.experimental.Accessors;
     @TemplateTransformation(value = "/manifest.vm", outputPath = "META-INF/services/io.fabric8.kubernetes.api.model.KubernetesResource", gather = true)
 })
 @Version("v1")
-@Group("cert-manager.io")
+@Group("acme.cert-manager.io")
 public class ChallengeList implements KubernetesResource, KubernetesResourceList<io.fabric8.certmanager.api.model.acme.v1.Challenge>
 {
 
@@ -83,7 +83,7 @@ public class ChallengeList implements KubernetesResource, KubernetesResourceList
      * 
      */
     @JsonProperty("apiVersion")
-    private String apiVersion = "cert-manager.io/v1";
+    private String apiVersion = "acme.cert-manager.io/v1";
     @JsonProperty("items")
     private List<io.fabric8.certmanager.api.model.acme.v1.Challenge> items = new ArrayList<io.fabric8.certmanager.api.model.acme.v1.Challenge>();
     /**

--- a/extensions/certmanager/model-v1/src/generated/java/io/fabric8/certmanager/api/model/acme/v1/Order.java
+++ b/extensions/certmanager/model-v1/src/generated/java/io/fabric8/certmanager/api/model/acme/v1/Order.java
@@ -70,7 +70,7 @@ import lombok.experimental.Accessors;
     @TemplateTransformation(value = "/manifest.vm", outputPath = "META-INF/services/io.fabric8.kubernetes.api.model.KubernetesResource", gather = true)
 })
 @Version("v1")
-@Group("cert-manager.io")
+@Group("acme.cert-manager.io")
 public class Order implements HasMetadata, Namespaced
 {
 
@@ -80,7 +80,7 @@ public class Order implements HasMetadata, Namespaced
      * 
      */
     @JsonProperty("apiVersion")
-    private String apiVersion = "cert-manager.io/v1";
+    private String apiVersion = "acme.cert-manager.io/v1";
     /**
      * 
      * (Required)

--- a/extensions/certmanager/model-v1/src/generated/java/io/fabric8/certmanager/api/model/acme/v1/OrderList.java
+++ b/extensions/certmanager/model-v1/src/generated/java/io/fabric8/certmanager/api/model/acme/v1/OrderList.java
@@ -73,7 +73,7 @@ import lombok.experimental.Accessors;
     @TemplateTransformation(value = "/manifest.vm", outputPath = "META-INF/services/io.fabric8.kubernetes.api.model.KubernetesResource", gather = true)
 })
 @Version("v1")
-@Group("cert-manager.io")
+@Group("acme.cert-manager.io")
 public class OrderList implements KubernetesResource, KubernetesResourceList<io.fabric8.certmanager.api.model.acme.v1.Order>
 {
 
@@ -83,7 +83,7 @@ public class OrderList implements KubernetesResource, KubernetesResourceList<io.
      * 
      */
     @JsonProperty("apiVersion")
-    private String apiVersion = "cert-manager.io/v1";
+    private String apiVersion = "acme.cert-manager.io/v1";
     @JsonProperty("items")
     private List<io.fabric8.certmanager.api.model.acme.v1.Order> items = new ArrayList<io.fabric8.certmanager.api.model.acme.v1.Order>();
     /**

--- a/extensions/certmanager/model-v1/src/main/resources/schema/certmanager-schema.json
+++ b/extensions/certmanager/model-v1/src/main/resources/schema/certmanager-schema.json
@@ -599,7 +599,7 @@
       "properties": {
         "apiVersion": {
           "type": "string",
-          "default": "cert-manager.io/v1",
+          "default": "acme.cert-manager.io/v1",
           "required": true
         },
         "kind": {
@@ -630,7 +630,7 @@
       "properties": {
         "apiVersion": {
           "type": "string",
-          "default": "cert-manager.io/v1",
+          "default": "acme.cert-manager.io/v1",
           "required": true
         },
         "items": {
@@ -719,7 +719,7 @@
       "properties": {
         "apiVersion": {
           "type": "string",
-          "default": "cert-manager.io/v1",
+          "default": "acme.cert-manager.io/v1",
           "required": true
         },
         "kind": {
@@ -750,7 +750,7 @@
       "properties": {
         "apiVersion": {
           "type": "string",
-          "default": "cert-manager.io/v1",
+          "default": "acme.cert-manager.io/v1",
           "required": true
         },
         "items": {

--- a/extensions/certmanager/model-v1/src/test/java/io/fabric8/certmanager/test/model/V1ChallengeTest.java
+++ b/extensions/certmanager/model-v1/src/test/java/io/fabric8/certmanager/test/model/V1ChallengeTest.java
@@ -1,0 +1,39 @@
+/**
+ * Copyright (C) 2015 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.fabric8.certmanager.test.model;
+
+import io.fabric8.certmanager.api.model.acme.v1.Challenge;
+import io.fabric8.certmanager.api.model.acme.v1.ChallengeBuilder;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+class V1ChallengeTest {
+
+  @Test
+  void testChallengeModelApiVersion() {
+
+    Challenge challenge = new ChallengeBuilder()
+        .withNewMetadata().withName("my-challenge").endMetadata()
+        .build();
+
+    assertNotNull(challenge);
+    assertEquals("my-challenge", challenge.getMetadata().getName());
+    assertEquals("acme.cert-manager.io/v1", challenge.getApiVersion());
+  }
+
+}

--- a/extensions/certmanager/model-v1/src/test/java/io/fabric8/certmanager/test/model/V1OrderTest.java
+++ b/extensions/certmanager/model-v1/src/test/java/io/fabric8/certmanager/test/model/V1OrderTest.java
@@ -1,0 +1,39 @@
+/**
+ * Copyright (C) 2015 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.fabric8.certmanager.test.model;
+
+import io.fabric8.certmanager.api.model.acme.v1.Order;
+import io.fabric8.certmanager.api.model.acme.v1.OrderBuilder;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+class V1OrderTest {
+
+  @Test
+  void testOrderModelApiVersion() {
+
+    Order order = new OrderBuilder()
+        .withNewMetadata().withName("my-order").endMetadata()
+        .build();
+
+    assertNotNull(order);
+    assertEquals("my-order", order.getMetadata().getName());
+    assertEquals("acme.cert-manager.io/v1", order.getApiVersion());
+  }
+
+}

